server/bin/pbench-audit-server.py

```diff
@@ -382,12 +383,30 @@ def output_format(self, f, verify_tardir, cnt):
         return cnt
 
 
-def verify_controllers(ihier, hier, hierarchy):
-    # cnt = 0
+def verify_hierarchy(ihier, hier, hierarchy):
+
     hierchy = os.path.basename(hierarchy)
```

Comment:
While I'm still working out the object structure here, this sort of context (essentially the meaning of the sub-hierarchy represented by the Hierarchy object) seems like something that should be encapsulated in the object rather than interpreted from outside the object as we're doing here. Even if this is the only place we make this distinction, it'd be cleaner to isolate the hierarchy semantics inside the object.

Since we already have IncomingHierarchy, ResultsHierarchy, and UserHierarchy subclasses, why are we essentially repeating the object type discrimination out here? It'd be cleaner if we could rely on them to know what they are and avoid exposing implementation details.

Incidentally, I think the name "hierchy" is an awkward abbreviation that's going to be hard to grasp for future readers of the code. Not terribly important since the name's scope is short and there are only a few references; but names that show the meaning of the variable always make the code easier to read, and natural words are better than unusual abbreviations. I think that with an expanded Hierarchy object that avoids this coupling, the variable just goes away; failing that, though, a name more like "hiertype" or "hierbase" would be easier on the eyes than "hierchy".

Reply:
Thank you @dbutenhof I am working on refactoring the code and your review helped a lot. I will work on these points and your further suggestions :)